### ISSUE TITLE
'Body materials report' issues

### ISSUE THREAD

**Issue by hoksilato2** [EDCD/EDDI#1029]
## EDDI version in which issue found
EDDI 3.3.0-rc2

## Steps to reproduce

1. script speech "Body materials report"
2. When you find planet with high level of any material

## Expected
reportMats and reportMatConc "Switches" will work like the description

## Observed
Only works when the planet has high level of materials although you have configured:

{set reportMats to 2} (spech all materials)
{set reportMatConc to 0} (only good and high %?)

## Investigation
Maybe it's related or it's another issue. 'material' variable is not translated. Ex: {goodmaterials[0].material}

but, in others scripts like "Material inventory report" the translate are correct. ex: {inventory.material}

**Comment by hoksilato2** [comment]
well I think that the key it's in {set reportMatConc to 0}. I need config material monitor for listen rare materials with 3% for example. Isn't it?

The Issue with material translation is more important :)

Thank you a lot.

**Comment by StivlWrith** [comment]
to have translated with material i use (in your exemple) {goodmaterials[0].definition.name}
Hope it can help you

**Comment by Tkael** [comment]
@richardbuckle There's a bit of wierdness with the code on this one... can you look into it? Ref. commit #b5ac3d51
The constructor in MaterialAmount.cs is as follows...
```cs
        public MaterialAmount(Material material, int amount)
        {
            Material My_material = Material.FromEDName(material.edname);
            this.material = My_material?.invariantName;
            this.edname = My_material?.edname;
            this.amount = amount;
            this.Category = My_material?.category.localizedName;
        }
```
It's not clear to me why we aren't using `this.material = My_material?.localizedName;`?

**Comment by Tkael** [comment]
You'll need to give a more specific example when it comes to the rare material. Which body did you scan and which material(s) did you expect to hear included that were not?

**Comment by Darkcyde13** [comment]
This wouldn't be something I've 'fixed' in my version of the script, would it? Referenced in issue #1050.  Just a thought.

**Comment by hoksilato2** [comment]
Thank you @darkcyde13!! I will use your script for reference. Realy, Thank you so much :)

**Comment by Darkcyde13** [comment]
I'm not totally sure on what your initial problem is, so I can't say 100% that my version of the script will fix the problem you are experiencing.  If it does, then great!

Let us know either way, because if it doesn't fix it, then it may need further investigation, and if it does fix it, then we know what needs to be done! :)

**Comment by qerk** [comment]
Hello! 

Had been struggling a bit with Body materials report not reporting anything if the event is Body scan.
Found this in Body report:
    {if reportMaterial: {F("Body materials report")} }

should be
    {if reportMaterials: {F("Body materials report")} }

**Comment by Darkcyde13** [comment]
@qerk - Yep that's one of my fixes as mentioned in #1050 ;)

**Comment by hoksilato2** [comment]
I Close this Issue because with @StivlWrith 's solution for translate materials, and @Darkcyde13 's script, I can work for a best script for spanish users. Thank you all!
